Implemented Move Vertex tool

sts1skj opened the pull request:
The new tool has six tool options, five of which are new:
- X Mirror
- Weld
- Merge
- Restrict To Surface
- Restrict To Plane
- Restrict To Normal

sts1skj committed:
Cleaned up tool option user interface code
The tool options are: x-mirror, connected-only, and all-selected-meshes (a.k.a.
global brush collision).
The main simplification was to consolidate all of the UI-maintenance code in
two functions.  In addition, I fixed a few things that were probably bugs:
- undiff, weight, color, and alpha had independent x-mirror options.
- all-selected-meshes would be toggled on when switching from the seg/part tab.
- x-mirror and connected-only were enabled for collapse-vertex, flip-edge,
	and split-edge
- connected-only was enabled on startup
- the text for all-selected-meshes was unclear

sts1skj committed:
Removed "Global Brush Collision" tool option
I also added a check to prevent the user from switching to the segments
or partitions tab with more than one mesh selected.

sts1skj committed:
Hopefully resolve lib/nifly conflict

sts1skj committed:
Implemented Move Vertex tool
The new tool has six tool options, five of with are new:
- X Mirror
- Weld
- Merge
- Restrict To Surface
- Restrict To Plane
- Restrict To Normal

sts1skj commented:
Some possibly unresolved stuff in this pull request:

I drew six new icons: one for the tool and the others for the tool options.  My icons are not necessarily good.

Should OutfitProject::IsVertexOnBoundary be in mesh instead and operate on mesh data instead of nif data?

I only added one key binding: Escape cancels an active vertex move.  Should there be more key bindings?

I made x-mirror mutually exclusive with weld and merge.  It would be messy to do a weld or merge with x-mirror on.  Should I implement it?

sts1skj committed:
Resolve nifly conflict

ousnius commented on src/program/OutfitProject.cpp:
```diff
@@ -3215,6 +3215,243 @@ bool OutfitProject::PrepareRefineMesh(NiShape* shape, UndoStateShape& uss, std::
+
+void OutfitProject::PrepareWeldVertex(NiShape* shape, UndoStateShape& uss, int selVert, int targVert, NiShape* targShape) {
+	// Get triangle data
+	uint16_t numVerts = shape->GetNumVertices();
```
@sts1skj numVerts is unused here.

sts1skj committed:
Deleted unused variable

ousnius commented on res/xrc/OutfitStudio.xrc:
```diff
@@ -229,6 +235,46 @@
+							<object class="tool" name="btnWeld">
+								<label>Weld Vertex</label>
+								<tooltip>Welds the selected vertex to the target vertex, giving it the target's coordinates, coordinate slider diffs, and bone weights, but not UV.</tooltip>
+								<bitmap>$(AppDir)\res\images\Weld.png</bitmap>
```
@sts1skj This file is missing from the commit.

sts1skj committed:
Added missing Weld.png

sts1skj committed:
Fixed some merge-vertex bugs
- s2 not initialized properly
- clicking okay in the message box in EndMoveVertex would cause onCaptureLost
to be called, which would call CancelMoveVertex.

sts1skj committed:
Updated merge and weld tooltips

sts1skj committed:
Made tool option buttons disappear instead of disable
Also moved tool options to the end of the vertical toolbar.

sts1skj committed:
Merge remote-tracking branch 'origin/dev' into moveVert

ousnius committed:
Enable tool items

ousnius committed:
Realize toolbar changes after every remove/insert

ousnius committed:
Apply clang format

sts1skj committed:
Made the weight brush have an independent x-mirror setting.
